core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java

```diff
@@ -112,13 +112,22 @@ public String tierAlias() {
     return mTierAlias;
   }
 
+  /**
+   * Check whether the location has no restriction, i.e., {@link #ANY_TIER_LOCATION}.
+   *
+   * @return true if location is equal to {@link #ANY_TIER_LOCATION}
```

Comment:
Looks like we hide the constant of ANY_TIER, ANY_DIR and ANY_MEDIUM behind these public methods, so can we make the constant be private?

Reply:
Made ANY_DIR and ANY_MEDIUM private. ANY_TIER is still used as a key in a map. The code looks hacky but will take more effort to clean up. Might do that later.